### PR TITLE
Additives explorer

### PR DESCRIPTION
**Pull request by PrajwalM2212**
#1330 

add

screenshot_1523902166
screenshot_1523902171

image

image

image

### CODE REVIEW

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/AdditivesAdapter.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+        public ViewHolder(View itemView) {
+            super(itemView);
+            additiveName = itemView.findViewById(R.id.additiveName);
+            itemView.setOnClickListener(this);
+        }
+
+        @Override
+        public void onClick(View view) {
+            clickListener.onclick(getAdapterPosition(),additives.get(getAdapterPosition()).getName());
```
Comment (ripetrescu) [inline app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/AdditivesAdapter.java]: Please rename the onclick method to onClick?

Also, in my opinion, the ViewHolder class, even if it's an inner class in this case, it shouldn't access directly AdditivesAdapter's fields like **additives** and **clickListener**. But a better approach will be something like:
`
package openfoodfacts.github.scrachx.openfood.views.adapters;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import java.util.List;

import openfoodfacts.github.scrachx.openfood.R;
import openfoodfacts.github.scrachx.openfood.models.AdditiveName;

/**
 * Created by prajwalm on 16/04/18.
 */

public class AdditivesAdapter extends RecyclerView.Adapter<AdditivesAdapter.ViewHolder> {

    private List<AdditiveName> additives;
    private ClickListener clickListener;

    public interface ClickListener {
        void onclick(int position, String name);
    }

    public AdditivesAdapter(List<AdditiveName> additives, ClickListener clickListener) {

        this.additives = additives;
        this.clickListener = clickListener;

    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext()).inflate(R.layout.additives_item, parent, false);
        return new ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {

        holder.additiveName.setText(additives.get(position).getName());
        holder.itemView.setOnClickListener(view -> {
            if (clickListener != null) {
                int pos = holder.getAdapterPosition();
                clickListener.onclick(pos, additives.get(pos).getName());
            }
        });
    }

    @Override
    public int getItemCount() {
        return additives.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {
        TextView additiveName;

        public ViewHolder(View itemView) {
            super(itemView);
            additiveName = itemView.findViewById(R.id.additiveName);
        }
    }
}
`
The benefit of this, is that you can reuse the same ViewHolder for a different Adapter if necessary, but it's your call :)

Comment (PrajwalM2212) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/adapters/AdditivesAdapter.java]: Thanks for letting me know of new ways to do things.

**Comment by teolemon** [comment]
``` 
:app:compileObfDebugJavaWithJavac/home/travis/build/openfoodfacts/openfoodfacts-androidapp/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java:37: error: AdditivesExplorer is not abstract and does not override abstract method onClick(int,String) in ClickListener
public class AdditivesExplorer extends BaseActivity implements AdditivesAdapter.ClickListener {
       ^
/home/travis/build/openfoodfacts/openfoodfacts-androidapp/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java:100: error: method does not override or implement a method from a supertype
    @Override
    ^
/home/travis/build/openfoodfacts/openfoodfacts-androidapp/app/src/main/java/openfoodfacts/github/scrachx/openfood/utils/WifiUploadReceiver.java:96: error: cannot find symbol
            String contentText = this.getResources().getQuantityString(R.plurals.offline_notification_count, listSaveProduct.size(), listSaveProduct.size());
                                                                                ^
  symbol:   variable offline_notification_count
  location: class plurals
Note: Some input files use or override a deprecated API.
Note: Recompile with -Xlint:deprecation for details.
Note: /home/travis/build/openfoodfacts/openfoodfacts-androidapp/app/src/main/java/openfoodfacts/github/scrachx/openfood/views/MainActivity.java uses unchecked or unsafe operations.
Note: Recompile with -Xlint:unchecked for details.
3 errors
 FAILED
FAILURE: Build failed with an exception.
```

File: app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            });
+
+        }
+
+
+        return super.onCreateOptionsMenu(menu);
+    }
+}
+
+
```
[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java]: I should not have forgotten this. Thanks 🙂

[the remark this answers is not in the document]
Comment (ripetrescu) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java]: Did you get a chance to update it?

[the remark this answers is not in the document]
Comment (PrajwalM2212) [reply app/src/main/java/openfoodfacts/github/scrachx/openfood/views/AdditivesExplorer.java]: @ripetrescu No. I have been busy these few days. I will update it soon.

**Comment by teolemon** [comment]
Thanks @ripetrescu for the review :-) 
Merging @PrajwalM2212

**Pull request merged**